packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -603,84 +603,37 @@ export class UiControlClient extends ApiCommands {
   async clickText(
     params: {
       text: string,
-      type: 'similar' | 'exact' | 'regex',
+      matching: TextMatchingOption,
       relation?: {
         type: RelationsForConvenienceMethods,
         text: string
       }
     },
   ) {
     let command: FluentFiltersOrRelations | FluentFiltersOrRelationsGetter = this.click().text();
-
-    command = await this.evaluateMatchingProperty(command, params.type, params.text);
-
+    command = this.evaluateMatchingProperty(
+      command,
+      { value: params.text, matching: params.matching },
+    );
     if (params.relation) {
       command = this.evaluateRelation(command, params.relation.type, params.relation.text);
     }
-
     await command.exec();
   }
 
   // eslint-disable-next-line class-methods-use-this
-  private async evaluateFinalExpectExistenceCommand(
-    finalCommand: FluentFiltersOrRelationsGetter,
-    element: ExpectExistenceInputParameter,
-  ): Promise<ExpectExistenceInputParameter> {
-    const el = element;
-    el.exists = false;
-    if ((await finalCommand.exec()).length > 0) {
-      el.exists = true;
-    }
-    return new Promise<ExpectExistenceInputParameter>((resolve) => {
-      resolve(el);
-    });
-  }
-
-  // eslint-disable-next-line class-methods-use-this
-  private async evaluateMatchingProperty(
-    command: FluentFiltersOrRelationsGetter | FluentFiltersOrRelations,
-    matching: 'similar' | 'exact' | 'regex' | undefined,
-    text: string,
-  ): Promise<FluentFiltersOrRelationsGetter | FluentFiltersOrRelations> {
-    let commando = command;
-    switch (matching) {
-      case 'similar':
-        commando = command.withText(text);
-        break;
+  private evaluateMatchingProperty(
+    command: FluentFiltersGetter | FluentFiltersOrRelationsGetter | FluentFiltersOrRelations,
+    text: ExpectExistenceInputParameterText,
+  ): FluentFiltersOrRelationsGetter | FluentFiltersOrRelations {
+    switch (text.matching ?? 'similar') {
       case 'exact':
-        commando = command.withExactText(text);
-        break;
+        return command.withExactText(text.value);
       case 'regex':
-        commando = command.withTextRegex(text);
-        break;
+        return command.withTextRegex(text.value);
       default:
-        throw new ValidationError('"matching" property must be "similar", "exact" or "regex"');
+        return command.withText(text.value);
```

Comment:
Why not
```typescript
case 'similar':
        return command.withText(text);
```
instead of the default.

Reply:
Actually, these are my reasons for changing it 😆 Parameter is optional and defaults to `similar`. Code currently implies that another value could be possible here as we have a general default clause which is not the case. If someone dropped out typechecking by TS and actually provided a different value, we would simply go with the default here instead of failing and educating the user with the error message. Let's say a user provided `'regexp'` instead of the valid `'regex'`, the user would expect that the regular expression is used instead we use similarity as we don't tell the user nothing about the mistake in using the api.